src/Psl/Iter/reduce.php

```diff
@@ -19,21 +19,21 @@
  *      Iter\reduce(Iter\range(1, 5), fn($accumulator, $value) => $accumulator * $value, 1)
  *      => 120
  *
- * @psalm-template Tk
+ * @psalm-template Tk of array-key
```

Comment:
iterable can have a non array-key keys.

e.g:

```
/**
 * @var iterable<list<int>, Collection\Vector<int>> $my_iterable
 */
$my_iterable = (static function(): iterable {
  yield [1, 2, 3] => new Collection\Vector([1, 2, 3]);
  yield [4, 5, 6] => new Collection\Vector([4, 5, 6]);
})();
```

Reply:
I did not know this was possible. On the other hand :  why?! :)